### PR TITLE
Add delta-stepping fusion algorithm

### PR DESCRIPTION
**Pull request by ddn0**

### CODE REVIEW

**Review by arthurp — CHANGES_REQUESTED** [review]
Can you add the new algorithm to the Python wrappers as well?

File: libgalois/src/analytics/sssp/sssp.cpp [hunk 1]
```diff
@@ -125,6 +127,106 @@ struct SsspImplementation : public katana::analytics::BfsSsspImplementationBase<
...
+    using Buckets = katana::gstl::Vector<Bucket>;
+
+    katana::PerThreadStorage<Buckets> buckets;
+
+    auto relax = [&](Node n, Dist sdist, Buckets& b) {
+      for (auto ii : graph->edges(n)) {
+        auto dest = graph->GetEdgeDest(ii);
+        auto& ddist = graph->template GetData<NodeDistance>(dest);
+        Dist ew = graph->template GetEdgeData<EdgeWeight>(ii);
+        const Dist new_dist = sdist + ew;
```
Comment (roshandathathri) [inline libgalois/src/analytics/sssp/sssp.cpp:145]: Drop `const`?

**Review by roshandathathri — APPROVED** [review]
LGTM

**Review by arthurp — APPROVED** [review]
LGTM.

**Pull request merged**